File: typescript/day02.test.ts
import * as load from './index'
import * as d from './day02'

describe('day 02', () => {
    jest.spyOn(load, 'loadInput').mockImplementation(async () => {
        return Promise.resolve(Buffer.from('A Y\nB X\nC Z'))
    })

    it('Part 1', async () => {
        await expect(d.partOne()).resolves.toEqual(15)
    })

    describe('part 1', () => {
        it('getOutcome', async () => {
            expect(d.getResponse('A', 'Y')).toBe(4)
        })

        it('Part 2', async () => {
            await expect(d.partTwo()).resolves.toEqual(12)
        })
    })
})

File: dart/test/day02_test.dart
import 'package:aoc2022/day02.dart';
import 'package:test/test.dart';

void main() {
  test('partone', () async {
    expect(await partone(), 12645);
  });

  test('parttwo', () async {
    expect(await parttwo(), 11756);
  });
}
